vernac/library.ml

```diff
@@ -152,17 +154,20 @@ let library_is_loaded dir =
 let register_loaded_library m =
   let libname = m.libsum_name in
   let rec aux = function
-    | [] ->
-        if (Global.typing_flags ()).enable_native_compiler then begin
-            let dirname = Filename.dirname (library_full_filename libname) in
-            Nativelib.enable_library dirname libname
-          end;
-        [libname]
+    | [] -> [libname]
     | m'::_ as l when DirPath.equal m' libname -> l
     | m'::l' -> m' :: aux l' in
   libraries_loaded_list := aux !libraries_loaded_list;
   libraries_table := DPmap.add libname m !libraries_table
 
+let register_native_library libname =
+  if (Global.typing_flags ()).enable_native_compiler
+    && not (DPset.mem libname !loaded_native_libraries) then begin
```

Comment:
Why do we need this new check on this new ref?

Reply:
I'm not 100% sure why the check is necessary, but it was already there (mixed in `register_loaded_library`). So I preserved the behavior that requiring a library that has already been loaded does not call `Nativelib.enable_library` again.

Reply:
ok I see it now